roles/lib_openshift/src/test/unit/test_oadm_manage_node.py

```diff
@@ -179,6 +180,114 @@ def test_schedulable_false(self, mock_openshift_cmd, mock_tmpfile_copy):
         self.assertEqual(results['results']['nodes'][0]['name'], 'ip-172-31-49-140.ec2.internal')
         self.assertEqual(results['results']['nodes'][0]['schedulable'], False)
 
+    @unittest.skipIf(six.PY3, 'py2 test only')
+    @mock.patch('os.path.exists')
+    @mock.patch('os.environ.get')
+    def test_binary_lookup_fallback(self, mock_env_get, mock_path_exists):
+        ''' Testing binary lookup fallback '''
+
+        mock_env_get.side_effect = lambda _v, _d: ''
+
+        mock_path_exists.side_effect = lambda _: False
+
+        self.assertEqual(ManageNode._locate_oc_binary(), 'oc')
```

Comment:
Where's the test testing that the order is relevant? It would have been a failing test.

Now that the purpose is more clear, isn't what we're trying to do augmenting/replacing the root's $PATH with the user's $PATH?

Security issues aside, I think that would've been easier to simple augment/replace env["PATH"] whenever we call a command with `sudo`. Are we talking about calls to external code from Python code? Is it code we control or part of Ansible?

If we need to have this "locate_oc" API for whatever reason, we may use `distutils.spawn.find_executable`, available in Py2/Py3, or simple equivalent code:

http://stackoverflow.com/a/33773405/4804690